torch/ao/quantization/backend_config/backend_config.py

```diff
@@ -63,6 +74,55 @@ class ObservationType(Enum):
     example: torch.cat, maxpool
     """
 
+
+@dataclass
+class DTypeWithConstraints:
+    """
+    Config for specifying additional constraints for a given dtype, such as quantization value
+    ranges and scale value ranges, to be used in :class:`~torch.ao.quantization.backend_config.DTypeConfig`.
+    """
+    dtype: torch.dtype
+    quant_min: Optional[float] = None
+    quant_max: Optional[float] = None
+    scale_min: Optional[float] = None
+    scale_max: Optional[float] = None
+
+    @classmethod
+    def from_dict(cls, dtype_with_constraints_dict: Dict[str, Any]) -> DTypeWithConstraints:
```

Comment:
since this is a new class, we do still need these to/from dict methods?

Reply:
Not sure, I just added these for completeness, otherwise we would lose the constraint information if a new user decides to call `to_dict`

Reply:
I removed these after making the Union type change. `dtype_config.to_dict()` should still work without this